Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -1320,41 +1336,58 @@ function vtkOpenGLTexture(publicAPI, model) {
     publicAPI.createTexture();
     publicAPI.bind();
 
-    // Source texture data from the PBO.
-    // model.context.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, true);
-    model.context.pixelStorei(model.context.UNPACK_ALIGNMENT, 1);
-
-    // Scale up the texture to the next highest power of two dimensions (if needed) and flip y.
     const needNearestPowerOfTwo =
       !model._openGLRenderWindow.getWebgl2() &&
       (!vtkMath.isPowerOfTwo(image.width) ||
         !vtkMath.isPowerOfTwo(image.height));
-    const canvas = document.createElement('canvas');
-    canvas.width = needNearestPowerOfTwo
-      ? vtkMath.nearestPowerOfTwo(image.width)
-      : image.width;
-    canvas.height = needNearestPowerOfTwo
-      ? vtkMath.nearestPowerOfTwo(image.height)
-      : image.height;
-
-    model.width = canvas.width;
-    model.height = canvas.height;
-
-    const ctx = canvas.getContext('2d');
-    ctx.translate(0, canvas.height);
-    ctx.scale(1, -1);
-    ctx.drawImage(
-      image,
-      0,
-      0,
-      image.width,
-      image.height,
-      0,
-      0,
-      canvas.width,
-      canvas.height
-    );
-    const safeImage = canvas;
+
+    let textureSource = image;
+    let targetWidth = image.width;
+    let targetHeight = image.height;
+
+    let flipY = true;
+
+    // For WebGL1, we need to scale the image to the nearest power of two
+    // dimensions if the image is not already a power of two. For WebGL2, we can
+    // use the image as is. Note: Chrome has a perf issue where the path
+    // HTMLImageElement -> Canvas -> texSubImage2D is faster than
+    // HTMLImageElement -> texSubImage2D directly. See
+    // https://issues.chromium.org/issues/41311312#comment7
+    // Tested on Chrome 137.0.7151.104 Windows 11
```

Comment:
Was this pathway evaluated beyond what was outlined in the linked issue?

Reply:
I am not sure, but as you can see the issue is from 2017(Chrome 57) and it's still valid til today.